holoviews/core/data/grid.py

```diff
@@ -354,7 +354,8 @@ def groupby(cls, dataset, dim_names, container_type, group_type, **kwargs):
                 group_data = group_data if np.isscalar(group_data) else group_data.columns()
             else:
                 group_data = cls.select(dataset, **select)
-            if np.isscalar(group_data):
+
+            if np.isscalar(group_data) or (len(dimensions) == dataset.ndims and len(dataset.vdims) == 1):
```

Comment:
FYI `np.isscalar` is always `False` on Dask Arrays. Not sure if these are Dask Arrays though.

Reply:
Right, that's the main thing I struggled with, I think the ``or`` branch here should handle that.

Reply:
So testing `group_data.shape == ()` should be valid for both.

Admittedly NumPy has some weird scalar like entities (e.g. 0-D Arrays), which would pass this test and fail `np.isscalar`. Though would suspect that distinction is irrelevant here. Would be interesting to know if that assumption is wrong.

Reply:
Thanks, I changed it. I also don't think that 0-D arrays would be relevant here, groupby should only produce scalars or ndarrays.